openmmtools/forcefactories.py

```diff
@@ -174,6 +174,240 @@ def restrain_atoms(thermodynamic_state, sampler_state, restrained_atoms, sigma=3
     thermodynamic_state.system = system
 
 
+from openmmtools.constants import ONE_4PI_EPS0
+default_energy_expression = "(4*epsilon*((sigma/r)^12-(sigma/r)^6) + k * chargeprod/r); sigma=0.5*(sigma1+sigma2); epsilon=sqrt(epsilon1*epsilon2); chargeprod=charge1*charge2; k={};".format(ONE_4PI_EPS0)
+
+
+def clone_nonbonded_parameters(nonbonded_force, 
+                               energy_expression=default_energy_expression,
+                               energy_prefactor=''):
+    """Creates a new CustomNonbonded force with the same global parameters,
+    per-particle parameters, and exception parameters as """
+
+    allowable_nb_methods = {openmm.NonbondedForce.CutoffPeriodic, openmm.NonbondedForce.CutoffNonPeriodic}
+    assert(nonbonded_force.getNonbondedMethod() in allowable_nb_methods)
+
+
+    # call constructor
+    # The 'energy_prefactor' allows us to easily change sign, or e.g. halve the energy if needed
+    new_force = openmm.CustomNonbondedForce(energy_prefactor+energy_expression)
+    new_force.addPerParticleParameter('charge')
+    new_force.addPerParticleParameter('sigma')
+    new_force.addPerParticleParameter('epsilon')
+
+
+    # go through all of the setter and getter methods
+    new_force.setCutoffDistance(nonbonded_force.getCutoffDistance())
+    #new_force.setEwaldErrorTolerance(nonbonded_force.getEwaldErrorTolerance())
+    #new_force.setForceGroup(nonbonded_force.getForceGroup())
+    new_force.setNonbondedMethod(nonbonded_force.getNonbondedMethod())
+    #new_force.setPMEParameters(*nonbonded_force.getPMEParameters())
+    #new_force.setReactionFieldDielectric(nonbonded_force.getReactionFieldDielectric())
+    #new_force.setReciprocalSpaceForceGroup(nonbonded_force.getReciprocalSpaceForceGroup())
+    new_force.setUseSwitchingFunction(nonbonded_force.getUseSwitchingFunction())
+    new_force.setSwitchingDistance(nonbonded_force.getSwitchingDistance())
+    new_force.setUseLongRangeCorrection(nonbonded_force.getUseDispersionCorrection())
+
+
+    # now add all the particle parameters
+    num_particles = nonbonded_force.getNumParticles()
+    for i in range(num_particles):
+        new_force.addParticle(nonbonded_force.getParticleParameters(i))
+
+    # now add all the exceptions? # TODO: check if we want to do this...
```

Comment:
I think you definitely want to copy all exceptions -> exclusions, or else you'll end up in NaNdyland.

Reply:
Will do!